includes/admin/class-wc-rest-payments-settings-controller.php

```diff
@@ -36,14 +46,16 @@ class WC_REST_Payments_Settings_Controller extends WC_Payments_REST_Controller {
 	/**
 	 * WC_REST_Payments_Settings_Controller constructor.
 	 *
-	 * @param WC_Payments_API_Client   $api_client WC_Payments_API_Client instance.
-	 * @param WC_Payment_Gateway_WCPay $wcpay_gateway WC_Payment_Gateway_WCPay instance.
-	 * @param WC_Payment_Gateways      $payment_gateways WC_Payment_Gateways instance.
+	 * @param WC_Payments_API_Client          $api_client WC_Payments_API_Client instance.
+	 * @param WC_Payment_Gateway_WCPay        $wcpay_gateway WC_Payment_Gateway_WCPay instance.
+	 * @param Digital_Wallets_Payment_Gateway $digital_wallets_gateway Digital_Wallets_Payment_Gateway instance.
+	 * @param WC_Payment_Gateways             $payment_gateways WC_Payment_Gateways instance.
 	 */
-	public function __construct( WC_Payments_API_Client $api_client, WC_Payment_Gateway_WCPay $wcpay_gateway, WC_Payment_Gateways $payment_gateways ) {
+	public function __construct( WC_Payments_API_Client $api_client, WC_Payment_Gateway_WCPay $wcpay_gateway, $digital_wallets_gateway, WC_Payment_Gateways $payment_gateways ) {
```

Comment:
Ah, got it, thanks for pointing that out! We could also instantiate [the controller here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments.php#L648-L650) only when the feature is enabled (which could also avoid [the check here](https://github.com/Automattic/woocommerce-payments/blob/add/1728-save-digital-wallets-settings/includes/admin/class-wc-rest-payments-settings-controller.php#L121-L124), since the controller would be added only when the feature is enabled).
I don't think the controller is used anywhere else where the feature is disabled.

Reply:
Sorry, can you update these 2 links please. I think the code base changed and I can't find the references.

Reply:
Sure thing! I updated the links in my comment 👍